src/app/pages/onboarding/magic-recovery-code/magic-recovery-code.tsx

```diff
@@ -1,81 +1,95 @@
 import { memo } from 'react';
-import { Text, Button, Input, Stack, StackProps } from '@stacks/ui';
+import { Formik, Form } from 'formik';
+import { useNavigate } from 'react-router-dom';
+import { Text, Button, Input, Stack, StackProps, Flex, Box } from '@stacks/ui';
 
 import { useRouteHeader } from '@app/common/hooks/use-route-header';
 import { useMagicRecoveryCode } from '@app/common/hooks/auth/use-magic-recovery-code';
-import { useMountEffect } from '@app/common/hooks/use-mount-effect';
 import { ErrorLabel } from '@app/components/error-label';
 import { Caption } from '@app/components/typography';
 import { Header } from '@app/components/header';
+import { RouteUrls } from '@shared/route-urls';
 
-const Form: React.FC<StackProps> = memo(props => {
-  const { onBack, onSubmit, onChange, magicRecoveryCodePassword, error, isLoading } =
-    useMagicRecoveryCode();
-
-  // Weird fix for preventing the input from using a value of the last input
-  // I think this is related to the routing and should be resolved with.
-  // https://github.com/blockstack/stacks-wallet-web/issues/1048
-  const mounted = useMountEffect();
+const MagicRecoveryCodeForm: React.FC<StackProps> = memo(props => {
```

Comment:
I merged these into just one component in the account list fix PR, do we really need two separate components just to separate a caption?

Reply:
In this example, sure, it's not hugely important. 

But generally speaking, we should be breaking components down more than we are, not less. Rather than looking at this as just the caption being different, I'd think of it as having one component responsible for containing the page and its content, and another for the form. If we add more elements later,  would these also just go into the same component? At what point does a component become too big, and have too many responsibilities?